backend/algorithms/objects/categories.py

```diff
@@ -48,17 +49,16 @@ def set_logic(self, logic: Logic):
         self.logic = logic
 
     def match_definition(self, course: str) -> bool:
-        if self.logic == Logic.AND:
-            return all(
-                [category.match_definition(course) for category in self.categories]
-            )
-        elif self.logic == Logic.OR:
-            return any(
-                [category.match_definition(course) for category in self.categories]
-            )
-        else:
+
+        if self.logic not in [Logic.AND, Logic.OR]:
             raise ValueError(f"Invalid logic: {self.logic}")
 
+        return all(
+            category.match_definition(course) for category in self.categories
+        ) if self.logic == Logic.AND else any(
+            category.match_definition(course) for category in self.categories
+        ) if self.logic == Logic.OR else None
+
```

Comment:
Personally I find this harder to read compared to Peter's branches changes, but maybe that's because I'm not used to this structure